Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -53,12 +53,15 @@ const double THRESHOLD = 1E-6;
 
 /** removes zeros from converged results
 * @param ws :: [input] The input workspace with zeros
-* @param maxIt :: [input] The max number of iteration this alg used
+* @param itCount [input] The number of iterations this alg used for each
+* spectrum
+* @param maxItCount :: [input] The max number of iteration this alg used
 * @param yLabel :: [input] y-label to use for returned ws
 * @return : ws cut down in lenght to maxIt
 */
-MatrixWorkspace_sptr removeZeros(const MatrixWorkspace_sptr &ws,
-                                 const size_t maxIt,
+MatrixWorkspace_sptr removeZeros(MatrixWorkspace_sptr &ws,
```

Comment:
@KarlPalmen thanks for this implementation and also improvements to same of the variable names used.

Looking at this method and where you have introduced resize() to adjust spectra length, I am wondering if there is a need for this method creating a new workspace? Instead could this method just resize the spectra of the existing input workspace? This should mean 1) avoid creation of an addition workspace 2) potentially less code

Reply:
The presence of separate input and output workspaces did confuse me and caused delay in making this change.